firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptState.swift

```diff
@@ -35,34 +35,50 @@ struct MicrosurveyPromptState: StateType, Equatable {
 
         switch action.actionType {
         case MicrosurveyPromptMiddlewareActionType.initialize:
-            let model = (action as? MicrosurveyPromptMiddlewareAction)?.microsurveyModel
-            return MicrosurveyPromptState(
-                windowUUID: state.windowUUID,
-                showPrompt: true,
-                showSurvey: false,
-                model: model
-            )
+            return handleInitializeAction(state: state, action: action)
         case MicrosurveyPromptActionType.closePrompt:
-            return MicrosurveyPromptState(
-                windowUUID: state.windowUUID,
-                showPrompt: false,
-                showSurvey: false,
-                model: state.model
-            )
+            return handleClosePromptAction(state: state)
         case MicrosurveyPromptActionType.continueToSurvey:
-            return MicrosurveyPromptState(
-                windowUUID: state.windowUUID,
-                showPrompt: true,
-                showSurvey: true,
-                model: state.model
-            )
+            return handleContinueToSurveyAction(state: state)
         default:
-            return MicrosurveyPromptState(
-                windowUUID: state.windowUUID,
-                showPrompt: state.showPrompt,
-                showSurvey: false,
-                model: state.model
-            )
```

Comment:
thanks @ionixjunior for the changes, looks good! I realize that we don't have tests for the default case, so if you want to add tests that would be great. This is an additional ask since we were missing tests to begin with. Let me know if you would like to add them. Otherwise, approved!

Reply:
@cyndichin thank you for pointing that out! I'll add the tests for the default case now.

Reply:
@cyndichin please check if this works. I've created a test for the default case by introducing a new invalid action type. This allows me to set an initial state and verify the results after calling the reducer. I would appreciate any feedback.

Reply:
@ionixjunior thank you for writing the tests 🥳 and improving our code coverage! Looks great, I added some small suggestions, let me know what you think!

Reply:
@cyndichin thank you for your suggestions! I've applied them and rebased the PR. I appreciate your help in improving this.

Reply:
thank you for the hard work + making the improvements @ionixjunior! It looks amazing now! There are some conflicts now, which may due to this [PR](https://github.com/mozilla-mobile/firefox-ios/pull/22812/files#diff-8aedc91f0f58a44d02f6d4d34b624a24b481491ec168534e1c2187ca053274df) that got recently merged. I can re-run bitrise and merge after addressing that.

Reply:
@cyndichin thank you so much! Your guidance on the other PR that introduced changes in this file was really helpful. I've updated and rebased to resolve the conflicts.